Sources/Test/SimpleTesting/Aggregates/HoppingWindowTopKAggregate.cs

```diff
@@ -0,0 +1,204 @@
+﻿// *********************************************************************
+// Copyright (c) Microsoft Corporation.  All rights reserved.
+// Licensed under the MIT License
+// *********************************************************************
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.StreamProcessing;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace SimpleTesting
+{
+    [TestClass]
+    public class HoppingWindowTopKAggregate : TestWithConfigSettingsAndMemoryLeakDetection
+    {
+        private const long HopSize = 10;
+        private const long WindowSize = 4 * HopSize;
+        private const int K = 3;
+
+        private StreamEvent<int> EndEvent = StreamEvent.CreatePunctuation<int>(StreamEvent.InfinitySyncTime);
+        private Random random = new Random(Seed: (int)DateTime.UtcNow.Ticks);
+
+        [TestMethod, TestCategory("Gated")]
+        public void TestHoppingWindowTopKAggregateSimple()
+        {
+            var input = new[]
+            {
+                StreamEvent.CreateStart(1, 10),
+                StreamEvent.CreateStart(3, 20),
+                StreamEvent.CreateStart(6, 10),
+                StreamEvent.CreateStart(8, 30),
+                StreamEvent.CreateStart(12, 20),
+                StreamEvent.CreateStart(18, 10),
+                EndEvent
+            };
+
+            var output = input.ToStreamable().HoppingWindowLifetime(10, 5).TopK(K);
+
+            var correctValues = new[] { 20, 30, 30, 20, 10 };
+            var correctEvents = new List<StreamEvent<int>>();
+            for (int i = 0; i < correctValues.Length; i++)
+            {
+                correctEvents.Add(StreamEvent.CreateStart(5 * (i + 1), correctValues[i]));
+                correctEvents.Add(StreamEvent.CreateEnd(5 * (i + 2), 5 * (i + 1), correctValues[i]));
+            }
+            correctEvents.Add(EndEvent);
```

Comment:
This is difficult to decipher. Please just list expected events explicitly. Currently this is difficult to debug.